### PR TITLE
[READY] Beaker of Holding

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This PR adds a new beaker for chemistry, the Beaker of Holding! Capable of holding up to 750 units, the beaker will aid Chemists and the like in being able to produce chemicals in bigger quantities than before! The research node for this is situated AFTER the Bluespace Storage node, aka the Bag of Holding. 

## Why It's Good For The Game

Hey kids, have you ever been too lazy to learn plumbing? And if you did are you too lazy to wait for it to do the job for you? Well here's the Beaker of Holding! For only the small price of a Bag of Holding (no this does not require an anomaly core), you can have you very own Beaker of Holding!

On a serious note, this beaker does NOT replace plumbing or make it obsolete. You still have to actually make and mix the chemicals, its not automatic. This will simply allow chemists to make medical chems and other reagents faster than they would with quantum or smaller beakers. Also these are not meant to be used in grenades, so dont worry we wont be having 2000u meth nades.

image
^one of the 4 sprites in its animation

## Changelog
:cl:
add: Added code for beaker
add: Added sprite for beaker
add: Added physical restriction in placing it in nades
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
You didn't blacklist these from chemical payloads.

**Comment by ghost** [comment]
> You didn't blacklist these from chemical payloads.

This is due to the fact that there is a chem bomb cap and you can already reach it with quantum beakers. Using holding beakers would be a waste of resources.

**Comment by ghost** [comment]
Beaker has been reduced to 750u and cannot be placed in any explosive except payloads.

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -254,20 +254,23 @@
...
 	ignition_temp = 500 // This is enough to expose a hotspot.
+	banned_containers =  list(/obj/item/reagent_containers/glass/beaker/beaker_of_holding)
 
 /obj/item/grenade/chem_grenade/adv_release // Intended for weaker, but longer lasting effects. Could have some interesting uses.
 	name = "advanced release grenade"
 	desc = "A custom made advanced release grenade. It is able to be detonated more than once. Can be configured using a multitool."
 	casedesc = "This casing is able to detonate more than once. Can be configured using a multitool."
 	icon_state = "timeg"
 	var/unit_spread = 10 // Amount of units per repeat. Can be altered with a multitool.
+	banned_containers =  list(/obj/item/reagent_containers/glass/beaker/beaker_of_holding)
```
Comment (Fikou) [inline code/game/objects/items/grenades/chem_grenade.dm]: dont

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 2]
```diff
@@ -254,20 +254,23 @@
...
 	ignition_temp = -100
+	banned_containers =  list(/obj/item/reagent_containers/glass/beaker/beaker_of_holding)
 
 /obj/item/grenade/chem_grenade/pyro // Intended for pyrotechnical mixes. Produces a small fire upon detonation, igniting potentially flammable mixtures.
 	name = "pyro grenade"
 	desc = "A custom made pyrotechnical grenade. Heats up contents upon ignition."
 	casedesc = "Upon ignition, it rapidly heats contents by 500°K."
 	icon_state = "pyrog"
 	ignition_temp = 500 // This is enough to expose a hotspot.
+	banned_containers =  list(/obj/item/reagent_containers/glass/beaker/beaker_of_holding)
```
Comment (Fikou) [inline code/game/objects/items/grenades/chem_grenade.dm]: dont do this either

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 3]
```diff
@@ -254,20 +254,23 @@
 	icon_state = "cryog"
 	affected_area = 2
 	ignition_temp = -100
+	banned_containers =  list(/obj/item/reagent_containers/glass/beaker/beaker_of_holding)
```
Comment (Fikou) [inline code/game/objects/items/grenades/chem_grenade.dm]: this not as well

File: code/game/objects/items/grenades/chem_grenade.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/grenades/chem_grenade.dm]: Im not sure what you mean here. Do you mean this is not how it should be written or that the beaking of holding shouldnt be banned from the nades?

Comment (Fikou) [reply code/game/objects/items/grenades/chem_grenade.dm]: you are overriding the parents banned containers to just ban beakers of holding, so now all grenades except basic ones would be able to fit in bluespace beakers, making large grenades useless

Comment (ghost) [reply code/game/objects/items/grenades/chem_grenade.dm]: ill add the bluespaces to the list

Comment (ghost) [reply code/game/objects/items/grenades/chem_grenade.dm]: actually no ill just remove the list entirely and let parent decide

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 4]
```diff
@@ -208,7 +208,6 @@
 	casedesc = "This casing affects a larger area than the basic model and can fit exotic containers, including slime cores and bluespace beakers. Heats contents by 25°K upon ignition."
 	icon_state = "large_grenade"
 	allowed_containers = list(/obj/item/reagent_containers/glass, /obj/item/reagent_containers/food/condiment, /obj/item/reagent_containers/food/drinks)
-	banned_containers = list()
```
Comment (Fikou) [inline code/game/objects/items/grenades/chem_grenade.dm:211]: you accidentally removed it from large grenades

**Comment by ATH1909** [comment]
Why do we need these when we already have T5 beakers (and barrels, for that matter)?

**Comment by ATH1909** [comment]
Splashing a beaker of holding that has 750u of itching powder in it onto someone will add enough itching powder to their system to deal around 7.5 points of liver damage to them per tick.

**Comment by ghost** [comment]
> Splashing a beaker of holding that has 750u of itching powder in it onto someone will add enough itching powder to their system to deal around 7.5 points of liver damage to them per tick.

You act as if thats a bad thing

**Pull request closed without merging**